src/EntryPoints/Specials/SpecialManageApprovers.php

```diff
@@ -99,10 +95,33 @@ private function renderHtml( array $approversCategories ): void {
 
 	/**
 	 * @param array<Approver> $approvers
+	 *
+	 * @return array<Approver>
+	 */
+	private function filterOutApproversWithNoCategories( array $approvers ): array {
+		$request = $this->getRequest();
+		$user = $this->userFactory->newFromName( $request->getText( 'username' ) );
+
+		return array_filter( $approvers, static function ( Approver $approver ) use ( $request, $user ) {
+			if ( !empty( $approver->categories ) ) {
+				return true;
+			}
+			return $request->wasPosted() && $approver->username === $user->getName();
+		} );
+	}
+
+	/**
+	 * @param array<Approver> $approvers
+	 *
 	 * @return array<array<string, mixed>>
 	 */
 	private function approversToViewModel( array $approvers ): array {
-		return array_map( fn( Approver $approver ) => $this->approverToViewModel( $approver ), $approvers );
+		return array_values(
```

Comment:
Why the added array_values? For the tests?

Reply:
This was causing an issue that @malberts reported above where deleting an older user made some rows disappear.

1- Our approversToViewModel method was returning an associative array where the keys were preserved from the original data structure. This resulted in an array like this:

```
array(1) {
  [1] => array(3) {
    ["username"] => string(10) "AdminName3"
    ["userId"] => int(4)
    ["categories"] => array(1) { ... }
  }
}
```

2- Problem with Mustache:
Mustache's {{#approvers}} loop expects an array with numeric indexes starting from 0. It doesn't work well with associative arrays or arrays with non-sequential numeric keys.

3- Solution:

- array_values() returns all the values from the array and indexes them numerically.
- This new structure is exactly what Mustache expects for its loops.
- It ensures that no matter how the original data is indexed, we always pass a cleanly indexed array to the template.

By adding array_values(), we convert the associative array into an indexed array which would print something like this:

```
array(1) {
  [0] => array(3) {
    ["username"] => string(10) "AdminName3"
    ["userId"] => int(4)
    ["categories"] => array(1) { ... }
  }
}
```

Reply:
Right, so we needed an integration test that looks at the output of mustache.

Turns out that building the HTML on the backend created several challenges. For instance, we now have a request that adds an approver with no categories that is then persisted, which on its own makes little sense, but is needed for the UI to show correctly. Perhaps building the UI with JS and sending background POSTs would have been better.